### PR TITLE
[script.subtitles.zimukux] 0.1.1

### PR DESCRIPTION
**Pull request by pizzamx**
### Add-on details:

- **General**
  - Add-on name: ZiMuKuX
  - Add-on ID: script.subtitles.zimukux
  - Version number: 0.1.1
  - Kodi/repository version: Matrix
- **Code location**
  - URL: https://github.com/pizzamx/zimuku_for_kodi
  - Revision/tag: v0.1.1
  - Branch: master

### Description of changes:
ZiMuKuX is a subtitle script for Kodi 19+ to download matching subtitle from ZiMuKu ( 字幕库 in Chinese, URL for now is [http://zimuku.org/](http://zimuku.org/)), originally forked from [https://github.com/taxigps/xbmc-addons-chinese/tree/master/service.subtitles.zimuku](https://github.com/taxigps/xbmc-addons-chinese/tree/master/service.subtitles.zimuku). This script is designed for Chinese users but anyone is welcomed to try.

### Checklist:
- [ ] My code follows the [add-on rules](http://kodi.wiki/view/Add-on_rules) and [piracy stance](http://kodi.wiki/view/Official:Forum_rules#Piracy_Policy) of this project. 
- [ ] I have read the [CONTRIBUTING](https://github.com/xbmc/repo-resources/blob/master/CONTRIBUTING.md) document
- [ ] Each add-on submission should be a single commit with using the following style: [plugin.video.foo] v1.0.0

### CODE REVIEW

File: script.subtitles.zimukux/resources/lib/sub_provider_service.py [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+        sel = xbmcgui.Dialog().select('请选择压缩包中的字幕', sub_name_list)
+        if sel == -1:
+            sel = 0
+        selected_sub = sub_file_list[sel]
+
+    logger.log(sys._getframe().f_code.co_name, "SUB FILE TO USE: %s" % selected_sub)
+    return [selected_sub]
+
+
+def get_params():
```
Comment (romanvm) [inline script.subtitles.zimukux/resources/lib/sub_provider_service.py:117]: The whole function can be replace with `dict(parse_qsl(sys.argv[2][1:]))`

File: script.subtitles.zimukux/resources/lib/sub_provider_service.py [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+            # no original title, get just Title
+            item['title'] = xbmc.getInfoLabel("VideoPlayer.Title")
+            # get movie title and year if is filename
+            if item['title'] == os.path.basename(xbmc.Player().getPlayingFile()):
+                title, year = xbmc.getCleanMovieTitle(item['title'])
+                item['title'] = title.replace('[', '').replace(']', '')
+                item['year'] = year
+
+        # Check if season is "Special"
+        if item['episode'].lower().find("s") > -1:
```
Comment (romanvm) [inline script.subtitles.zimukux/resources/lib/sub_provider_service.py:167]: `if 's' in item['episode'].lower()`

File: script.subtitles.zimukux/resources/lib/sub_provider_service.py [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+            #
+            item['season'] = "0"
+            item['episode'] = item['episode'][-1:]
+
+        if (item['file_original_path'].find("http") > -1):
```
Comment (romanvm) [inline script.subtitles.zimukux/resources/lib/sub_provider_service.py:172]: The same: `if 'http' in item['file_original_path']`

**Review by romanvm — CHANGES_REQUESTED** [review]
Please check the comments to `addon.xml` and the one about a hardcoded UI string. Other comments are not mandatory.

**Comment by pizzamx** [comment]
Please review this commit, which has addressed the issues

**Comment by romanvm** [comment]
You still haven't fixed a hardcoded UI string in `sub_provider_service.py:106`. Please move the string to translation files.

**Review by romanvm — APPROVED** [review]
LGTM now

**Comment by pizzamx** [comment]
Anything else I should submit or action I should take? @romanvm

**Review by romanvm — APPROVED** [review]
LGTM

**Pull request merged**